crypto/message.py

```diff
@@ -0,0 +1,58 @@
+from binascii import hexlify, unhexlify
+from hashlib import sha256
+
+from ecdsa import SECP256k1, SigningKey, VerifyingKey
+from ecdsa.util import sigdecode_der, sigencode_der_canonize
+
+from crypto.identity.keys import (
+    privat_key_from_secret, public_key_from_secret, uncompress_ecdsa_public_key
+)
+
+
+def sign_message(message, secret):
+    """Sign a message
+
+    Args:
+        message (bytes): a message you want to signature
+        secret (bytes): passphrase you want to use to sign the message
+
+    Returns:
+        dict: dict containing message, public_key and a signature data
+    """
+    private_key = privat_key_from_secret(secret)
+    signin_key = SigningKey.from_string(unhexlify(private_key), SECP256k1)
+    signature = hexlify(
+        signin_key.sign_deterministic(message, hashfunc=sha256, sigencode=sigencode_der_canonize)
+    )
+    data = {
+        'message': message,
+        'public_key': public_key_from_secret(secret),
+        'signature': signature,
+    }
+    return data
+
+
+def verify_message(message, public_key, signature):
```

Comment:
IMO it would be better that this would raise an exception if the key is invalid, rather than return True of False. That way, it corresponds to the name `verify_message`.

Reply:
should I just leave the ecdsa exception to be raised and not capture anything?

Reply:
I would handle it and raise a new exception, just so when you're working with this lib, you always import exceptions from `client.exceptions` or wherever they live, instead of looking for ecdsa exceptions.